File: AtlasWeb/Core/Modules/Tibbi_Surec/Geleneksel_Alternatif_Tamamlayici_Uygulamalar_Modulu/GetatExaminationFormViewModel.ts
//$7A9BFE02
import { BaseViewModel } from "NebulaClient/Model/BaseViewModel";
import { TraditionalMedicine } from "NebulaClient/Model/AtlasClientModel";
import { TraditionalMedAppRegion } from "NebulaClient/Model/AtlasClientModel";
import { TradititionalMedAppCases } from "NebulaClient/Model/AtlasClientModel";
import { SKRSGETATUygulamaTuru } from "NebulaClient/Model/AtlasClientModel";
import { SKRSGETATUygulamaBolgesi } from "NebulaClient/Model/AtlasClientModel";
import { SKRSGETATUygulandigiDurumlar } from "NebulaClient/Model/AtlasClientModel";
import { SKRSGETATUygulamaBirimi } from "NebulaClient/Model/AtlasClientModel";
import { SKRSGETATTedaviSonucu } from "NebulaClient/Model/AtlasClientModel";
import { Type } from "app/NebulaClient/ClassTransformer";

export class GetatExaminationFormViewModel extends BaseViewModel {
    protected __type__: string = 'Core.Models.GetatExaminationFormViewModel, Core';

    
    @Type(() => TraditionalMedicine)
    public _TraditionalMedicine: TraditionalMedicine = new TraditionalMedicine();

    @Type(() => TraditionalMedAppRegion)
    public TraditionalMedAppRegionGridList: Array<TraditionalMedAppRegion> = new Array<TraditionalMedAppRegion>();
    
    @Type(() => TradititionalMedAppCases)
    public TraditionalMedAppCasesGridList: Array<TradititionalMedAppCases> = new Array<TradititionalMedAppCases>();

    @Type(() => SKRSGETATUygulamaTuru)
    public SKRSGETATUygulamaTurus: Array<SKRSGETATUygulamaTuru> = new Array<SKRSGETATUygulamaTuru>();

    @Type(() => SKRSGETATUygulamaBolgesi)
    public SKRSGETATUygulamaBolgesis: Array<SKRSGETATUygulamaBolgesi> = new Array<SKRSGETATUygulamaBolgesi>();

    @Type(() => SKRSGETATUygulandigiDurumlar)
    public SKRSGETATUygulandigiDurumlars: Array<SKRSGETATUygulandigiDurumlar> = new Array<SKRSGETATUygulandigiDurumlar>();

    @Type(() => SKRSGETATUygulamaBirimi)
    public SKRSGETATUygulamaBirimis: Array<SKRSGETATUygulamaBirimi> = new Array<SKRSGETATUygulamaBirimi>();

    @Type(() => SKRSGETATTedaviSonucu)
    public SKRSGETATTedaviSonucus: Array<SKRSGETATTedaviSonucu> = new Array<SKRSGETATTedaviSonucu>();
}

File: AtlasWeb/Core/Modules/Tibbi_Surec/Geleneksel_Alternatif_Tamamlayici_Uygulamalar_Modulu/GetatExaminationFormViewModel.cs
//$8977512B
using System;
using System.Linq;
using Core.Models;
using Infrastructure.Filters;
using Infrastructure.Models;
using TTInstanceManagement;
using TTObjectClasses;
using TTDefinitionManagement;
using TTUtils;

namespace Core.Controllers
{
    public partial class TraditionalMedicineServiceController
    {
  
    }
}

namespace Core.Models
{
    public partial class GetatExaminationFormViewModel : ISpecialityBasedObjectViewModel
    {
        public void AddSpecialityBasedObjectViewModelToContext(TTObjectContext objectContext)
        {
            var tradMed = (TraditionalMedicine)objectContext.AddObject(this._TraditionalMedicine);

            if (this.TraditionalMedAppRegionGridList != null)
            {
                foreach (var item in this.TraditionalMedAppRegionGridList)
                {
                    var traditionalMedAppRegionImported = (TraditionalMedAppRegion)objectContext.AddObject(item);
                    if (((ITTObject)traditionalMedAppRegionImported).IsDeleted)
                        continue;
                    traditionalMedAppRegionImported.TraditionalMedicine = tradMed;
                }
            }

            if (this.TraditionalMedAppCasesGridList != null)
            {
                foreach (var item in this.TraditionalMedAppCasesGridList)
                {
                    var traditionalMedAppCasesImported = (TradititionalMedAppCases)objectContext.AddObject(item);
                    if (((ITTObject)traditionalMedAppCasesImported).IsDeleted)
                        continue;
                    traditionalMedAppCasesImported.TraditionalMedicine = tradMed;
                }
            }
        }
    }
}
